code/renderers/vue3/src/testing-api.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  composeStory as originalComposeStory,
+  composeStories as originalComposeStories,
+  setProjectAnnotations as originalSetProjectAnnotations,
+} from '@storybook/preview-api';
+import type {
+  Args,
+  ProjectAnnotations,
+  StoryAnnotationsOrFn,
+  Store_CSFExports,
+  StoriesWithPartialProps,
+} from '@storybook/types';
+
+import * as defaultProjectAnnotations from './render';
+import type { Meta } from './public-types';
+import type { VueRenderer } from './types';
+
+/** Function that sets the globalConfig of your Storybook. The global config is the preview module of your .storybook folder.
+ *
+ * It should be run a single time, so that your global config (e.g. decorators) is applied to your stories when using `composeStories` or `composeStory`.
+ *
+ * Example:
+ *```jsx
+ * // setup.js (for jest)
+ * import { setProjectAnnotations } from '@storybook/vue3';
+ * import projectAnnotations from './.storybook/preview';
+ *
+ * setProjectAnnotations(projectAnnotations);
+ *```
+ *
+ * @param projectAnnotations - e.g. (import projectAnnotations from '../.storybook/preview')
+ */
+export function setProjectAnnotations(
+  projectAnnotations: ProjectAnnotations<VueRenderer> | ProjectAnnotations<VueRenderer>[]
+) {
+  originalSetProjectAnnotations<VueRenderer>(projectAnnotations);
+}
+
+/**
+ * Function that will receive a story along with meta (e.g. a default export from a .stories file)
+ * and optionally projectAnnotations e.g. (import * from '../.storybook/preview)
+ * and will return a composed component that has all args/parameters/decorators/etc combined and applied to it.
+ *
+ *
+ * It's very useful for reusing a story in scenarios outside of Storybook like unit testing.
+ *
+ * Example:
+ *```jsx
+ * import { render } from '@testing-library/vue';
+ * import { composeStory } from '@storybook/vue3';
```

Comment:
However it is time to fix the render function in Vue render to match the official test api, there is a bit of issue here
I find that the testing-library/vue is more legit that the render function in @storybook/vue3 
Also i suggest to rename @storybook/vue3 , @storybook/vue, to match with vue current naming convention. and rename @storybook/vue to @storybook/vue2.7 . we need to be consistent with other api so users won't be confused 

NB:Dropping Vue 2.7 is recommended